assignment/scripts/array-practice.js

```diff
@@ -1,94 +1,123 @@
-console.log('Added an animal to end,', animalArray);
+animalArray.push("penguin");
+console.log("Added an animal to end,", animalArray);
 
 // 4.a. TODO: Add a new food at the end of your array & log the array
-
-
+favoriteFoods.push("Tacos");
+console.log("Added tacos to end", favoriteFoods);
 
 // Example: Remove the last animal by using Array.pop
 let removedAnimal = animalArray.pop();
-console.log('Removed the last animal', removedAnimal);
-console.log('The animals are now', animalArray);
+console.log("Removed the last animal", removedAnimal);
+console.log("The animals are now", animalArray);
 
-// 4.b. TODO: Remove the food at the end of your array & 
+// 4.b. TODO: Remove the food at the end of your array &
 //      log both the food removed and the updated array
-
+let removedFood = favoriteFoods.pop();
+console.log("Removed the last food", removedFood);
+console.log("The foods are now", favoriteFoods);
 
 // Example: Add an animal to the beginning using Array.unshift
-animalArray.unshift('walrus');
+animalArray.unshift("walrus");
 console.log(`Added an animal to beginning: ${animalArray}`);
 
 // 4.c. TODO: Add a food at the beginning of the array & log the array
-
+favoriteFoods.unshift("Tacos");
+console.log(`Added Tacos at the beginning of the array`);
 
 // Example: Remove the first animal using Array.shift
 removedAnimal = animalArray.shift();
-console.log('Removed the first animal', removedAnimal);
-console.log('The animals are now', animalArray);
+console.log("Removed the first animal", removedAnimal);
+console.log("The animals are now", animalArray);
 
-// 4.d TODO: Remove the food at the beginning of your array & 
+// 4.d TODO: Remove the food at the beginning of your array &
 //     log both the food removed and the updated array
+removedFood = favoriteFoods.shift();
+console.log("Removed the first food", removedFood);
+console.log("The foods are now", favoriteFoods);
 
 // 4.e (STRETCH) TODO: Replace the second food in your array
 //      with another one of your favorite foods.
 //      Then log the updated array.
+favoriteFoods[1] = "Sushi";
 
 // 4.f (STRETCH) TODO: Sort your favoriteFoods array
 //     in reverse alphabetical order. Log the array.
+const reverseFoods = favoriteFoods
+  .map((food) => food.toLowerCase())
+  .sort()
+  .reverse();
+console.log(reverseFoods);
 
 // 4.g (STRETCH) TODO: Convert your array to a string
 //     putting the word "and" between each item.
 //     eg "tacos and pizza and pasta". Log the string.
 
-// 4.h (STRETCH) TODO: Make a new array that combines 
+// title case function > "raspberry pie" => "Raspberry Pie"
+
+const toTitleCase = (string) => {
+  const words = string.trim().split(" ");
+  const titleWords = words.map((word) => {
+    const trimmedWord = word.trim();
+    return trimmedWord[0].toUpperCase() + trimmedWord.slice(1).toLowerCase();
+  });
+
+  return titleWords.join(" ");
+};
+
+const titleFavoriteFoods = favoriteFoods.map((food) => toTitleCase(food));
```

Comment:
Could also have done:
```suggestion
const titleFavoriteFoods = favoriteFoods.map(toTitleCase);
```

🤯